app/src/main/java/fr/coppernic/scansample/AndroidInteractor.java

```diff
@@ -6,15 +6,38 @@
 import java.util.List;
 
 class AndroidInteractor {
+    private static final String BASE_NAME_SYSTEM_SERVICE = "fr.coppernic.service";
+    private static final String BASE_NAME_BARCODE_MANAGER = "fr.coppernic.features.barcode";
+    private Context context;
 
-    String isAppInstalled(final Context context, final String packageName) {
+
+    AndroidInteractor(Context ctx) {
+        context = ctx;
+    }
+
+    String loadPackage() {
         final List<ApplicationInfo> appsInfo = context.getPackageManager().getInstalledApplications(0);
         for (final ApplicationInfo appInfo : appsInfo) {
-            if (appInfo.packageName.startsWith(packageName)) {
+            if (appInfo.packageName.startsWith(BASE_NAME_BARCODE_MANAGER)) {
```

Comment:
pour eviter 2 fois la boucle for, tu peux fair un startwith(MANAGER) || startwith(SYSTEM_SERVICE)

Reply:
c'est pas faux...